### PR TITLE
Add stack-debugging features using Abseil

### PR DESCRIPTION
**Pull request by veblush**
This adds following things to the test utility
- Modifying `grpc_test_init` to 
  - Install Abseil FailureSignalHandler instead of gRPC in-house one so that we can free from maintaining it
  - Call InitializeSymbolizer so that CurrentStackTrace can be used throughout the tests.
- Adding `CurrentStackTrace` so that stacktrace can be examined throughout the tests.

This feature is disabled on Windows because of #24178.

### CODE REVIEW

**Comment by veblush** [comment]
Failing tests in Bazel RBE Windows 

- //test/core/client_channel/resolvers:dns_resolver_connectivity_using_native_resolver_test
- //test/cpp/qps:qps_json_driver_test_cpp_generic_async_streaming_qps_unconstrained_insecure
- //test/cpp/qps:qps_json_driver_test_cpp_protobuf_async_streaming_from_client_1channel_1MB
- //test/cpp/qps:qps_json_driver_test_cpp_protobuf_sync_streaming_qps_unconstrained_insecure

```
[symbolize_win32.inc : 53] RAW: SymInitialize() failed: 2147483653
```

2147483653 = 0x80000005

**Comment by veblush** [comment]
Once the name of some tests were shortened, failing test has changed

- //test/cpp/qps:qps_jd_cpp_generic_async_streaming_qps_unconstrained_2waysharedcq_insecure
- //test/cpp/qps:qps_jd_cpp_protobuf_async_streaming_from_client_qps_unconstrained_insecure
- //test/cpp/qps:qps_jd_cpp_protobuf_async_streaming_from_server_qps_unconstrained_insecure
- //test/cpp/qps:qps_jd_cpp_protobuf_async_streaming_qps_unconstrained_2waysharedcq_insecure

**Comment by yashykt** [comment]
Can you post a sample output?

File: test/core/util/examine_stack.h [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+#include <grpc/support/port_platform.h>
+
+#include <string>
+
+namespace grpc_core {
+
+// Return the current stack trace as a string (on multiple lines, beginning with
+// "Stack trace:\n")
+std::string CurrentStackTrace();
```
Comment (yashykt) [inline test/core/util/examine_stack.h:30]: is this available on all platforms?

**Comment by veblush** [comment]
Sample output is

```
Stack trace:
    @     0x7f07d608aee4        192  grpc_test_init()
    @     0x7f07d608b0ff         48  grpc::testing::TestEnvironment::TestEnvironment()
    @           0x403856         64  main
    @     0x7f07d4a8ecca  (unknown)  __libc_start_main
    @ 0x5541d68949564100  (unknown)  (unknown)
```

File: test/core/util/examine_stack.h (hunk 1 above)
Comment (veblush) [reply test/core/util/examine_stack.h:30]: Yes, it is.

Comment (yashykt) [reply test/core/util/examine_stack.h:30]: This is not being used as of now. Can we modify some test to print out the stack trace?

**Review by yashykt — APPROVED** [review]

File: test/core/util/examine_stack.h (hunk 1 above)
Comment (veblush) [reply test/core/util/examine_stack.h:30]: Sure. I'm going to modify the test environment to make use of this in a following PR.

**Pull request merged**